### ISSUE TITLE
expanded property does not work

### ISSUE THREAD

**Issue by mohahn** [vitmalina/w2ui#1693]
* **I'm submitting a ...**
[X] bug report
[ ] feature request
[ ] question about the decisions made in the repository

When using exactly the source code from example http://w2ui.com/web/demos/#!grid/grid-7 and adding `expanded: true` to the records object, the rendered table does show the toggled expand icon, but the child rows are not displayed. When clicking on toggle icon an JS error is shown in console (using nightly JS):

```
w2ui.js?v=0.01:7349 Uncaught TypeError: Cannot read property 'expanded' of undefined
    at clearExpanded (w2ui.js?v=0.01:7349)
    at w2grid.collapse (w2ui.js?v=0.01:7298)
    at w2grid.toggle (w2ui.js?v=0.01:7205)
    at HTMLSpanElement.onclick (csadmin:1)
```
After second click on toggle icon the expanding and collapsing is working. But the problem is, that initial expanding on rendering does not work. This occurs with JS from example source, and with 1.5RC1 and nightly also.

**Comment by mpf82** [comment]
Parents can not be expanded by simply setting ``expanded`` to true. 
The children are not shown because they are not known in ``w2grid.records`` (they are merly an additional property on the parent).
Please look at the code of ``w2grid.expand``.

**Comment by mohahn** [comment]
http://w2ui.com/web/docs/1.5/w2grid.records does *not* state that `expanded` cannot be used when defining records. Why can't w2ui simply trigger expand method itself, when defined in records object?

**Comment by aurora** [comment]
I have the same issue. Is there a workaround for it?

**Comment by hero9** [comment]
Also tried to use `expanded:true` property inside record object, doesn't work. [How to use expanded property](http://w2ui.com/web/docs/1.5/w2grid.records)

Tried to expand all records where children length more than zero, by invoking `w2ui.grid.expand(recid)`. Also doesn't work, Error:
```
w2ui-1.5.rc1.min.js:6 Uncaught TypeError: Cannot read property 'w2ui' of undefined
```

@vitmalina @mpf82 guys, could you help me with expand records on render?

Thanks in advance!

**Comment by vitmalina** [comment]
Try `w2ui: { expanded: true }`

**Comment by hero9** [comment]
If you mean write it inside records object, then I’ve already tried it. It doesn’t work also. Although the expand arrow in expanded position.

Example of my record object with children:
{ recid:1, path:””, type:””,  w2ui: { expanded: true, children: [ ... ] } }

**Comment by vitmalina** [comment]
Yes, looks like you are right. I looked into the code and it does not see to be clear how to fix it. For now, use the work around of calling .expand() after the grid has been rendered.

**Comment by hero9** [comment]
As I said above it doesn't work. The error is:
```
w2ui-1.5.rc1.min.js:6 Uncaught TypeError: Cannot read property 'w2ui' of undefined
```

**Comment by mpf82** [comment]
Workaround / suggestion for implementation:

    /**
     * Expands all records of a given grid recursively if the records have children.
     *
     * @param {w2grid} grid - w2ui grid object
     * @returns {void}
     */
    function expand_all ( grid ) {
        // expand all records with children recursively
        var do_expand = true;
        while ( do_expand ) {
            var records_count = grid.records.length;
            grid.records.forEach( ( rec ) => {
                if ( rec.w2ui && rec.w2ui.children ) {
                    grid.expand( rec.recid );
                }
            } );
            do_expand = ( records_count !== grid.records.length );
        }
    }

**Comment by ajbrickhouse** [comment]
> Workaround / suggestion for implementation:
> 
> ```
> /**
>  * Expands all records of a given grid recursively if the records have children.
>  *
>  * @param {w2grid} grid - w2ui grid object
>  * @returns {void}
>  */
> function expand_all ( grid ) {
>     // expand all records with children recursively
>     var do_expand = true;
>     while ( do_expand ) {
>         var records_count = grid.records.length;
>         grid.records.forEach( ( rec ) => {
>             if ( rec.w2ui && rec.w2ui.children ) {
>                 grid.expand( rec.recid );
>             }
>         } );
>         do_expand = ( records_count !== grid.records.length );
>     }
> }
> ```

This works for me. Thank you.
